docs/fundamentals/flow/create-flow.md

```diff
@@ -675,6 +675,76 @@ Flow with 3 replicas of slow_encoder and 1 replica of fast_indexer
 The above Flow will create a topology with three Replicas of Executor `slow_encoder`. The `Flow` will send every 
 request to exactly one of the three instances. Then the replica will send its result to `fast_indexer`.
 
+
+### Replicate on multiple GPUs
+
+You can be in a setup when you want to replicate your Executor so that each replicas can use a different GPU on your machine.
+To do so you need to tell the Flow to leverage multiple GPUs, by passing `CUDA_VISIBLE_DEVICES=RR` as an environment variable,
+the Flow will then assign each available GPU to each replicas in a round-robin fashion.
+
+```{caution} 
+Replicate on multiple GPUs by using `CUDA_VISIBLE_DEVICES=RR` should only be used locally.  
+```
+
+```{tip}
+When working in Kubernetes or with Docker Compose you shoud allocate GPU ressources to each replica directly in the configuration files.
+```
+
+For example, if you have 3 GPUs and one of your Executor has 5 replicas then 
+
+````{tab} Python
+ In a `flow.py` file 
+```python
+from jina import Flow
+
+with Flow().add(
+    uses='jinahub://CLIPEncoder', replicas=5, install_requirements=True
+) as f:
+    f.block()
+```
+
+```shell
+CUDA_VISIBLE_DEVICES=RR python flow.py
+```
+````
+
+````{tab} YAML
+In a `flow.yaml` file
+```yaml
+jtype: Flow
+executors:
+- uses: jinahub://CLIPEncoder
+  install_requirements: True
+  replicas: 5  
+```
+
+```shell
+CUDA_VISIBLE_DEVICES=RR jina flow --uses flow.yaml
+```
+````
+
+The Flow will assign GPU devices in the following round-robin fashion:
+
+| GPU device | Replica ID |
+|------------|------------|
+| 0          | 0          |
+| 1          | 1          |
+| 2          | 2          |
+| 0          | 3          |
+| 1          | 4          |
+
+ 
+You can also restrict the visible devices in round-robin assignment by `CUDA_VISIBLE_DEVICES=RR0:2`, where `0:2` has the same meaning as Python slice. This will create the following assignment:
```

Comment:
Although this is not documented in the clip as service documentation, I believe it's good to let users know that they might need to separate the allocation of devices to executors.
you can check how this separation is implemented in [dalle flow](https://github.com/jina-ai/dalle-flow/blob/main/flow.yml#L41).
In general, users should be aware that they can use this feature in order to allocate 1 device for exec 1, RR1:3 for a replicated executor 2, etc. This can help them avoid issues like many processes trying to access the same resource

Reply:
it is documented in clip as as service : https://clip-as-service.jina.ai/user-guides/server/?utm_source=clip-as-service#serving-on-multiple-gpus.

You think that the current documentation on the round robin things is not enough ? What would you like to add ?

Reply:
For me, restricting the gpus for the entire flow is not the same as restricting gpus for some executors. The feature can also serve as a protection mechanism against executors accessing the same resource (and htat's how we are trying to use it currently).
What I mean is giving an example where a user can do:
```yaml

  - name: Exec1
    ...
    env:
      CUDA_VISIBLE_DEVICES: 0

  - name: Exec2
    replicas: 2
    ...
    env:
      CUDA_VISIBLE_DEVICES: RR1:3

  - name: Exec3
    ...
    env:
      CUDA_VISIBLE_DEVICES: 3:5
```

Reply:
Okay now I understand your point